data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -99,8 +100,17 @@ public KeyValueProcessor(final PluginMetrics pluginMetrics, final KeyValueProces
             throw new PatternSyntaxException("delete_value_regex is not a valid regex string", keyValueProcessorConfig.getDeleteValueRegex(), -1);
         }
 
-        if (keyValueProcessorConfig.getIncludeKeys() != null) {
-            includeKeysSet.addAll(keyValueProcessorConfig.getIncludeKeys());
+        includeKeysSet.addAll(keyValueProcessorConfig.getIncludeKeys());
+        excludeKeysSet.addAll(keyValueProcessorConfig.getExcludeKeys());
+
+        if (includeKeysSet.equals(excludeKeysSet) && !includeKeysSet.isEmpty()) {
```

Comment:
I think this check is still unnecessary.

Reply:
Also as best practice, it is more readable to encapsulate L106-L114 into a separate private method `validateKeySets()` rather than expose the validation details in the constructor.